### PR TITLE
Andree/superagent to master

### PR DESCRIPTION
**Pull request by AndreeFrank**
Have switched requestPromise to superagent for request handling

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,8 +6,11 @@
     "@types/jest": "24.0.12",
     "@types/node": "12.7.2",
     "@types/request-promise": "4.1.44",
```
Comment (toshi38) [inline package.json]: I guess we can remove request-promise and its types now?

File: package.json [hunk 2]
```diff
@@ -6,8 +6,11 @@
...
+    "@types/superagent": "4.1.3",
     "request": "2.88.0",
```
Comment (toshi38) [inline package.json]: request can go as well I guess?

File: src/gaus-client.ts [hunk 3]
```diff
@@ -92,26 +100,31 @@ export class GausClient {
...
+      .set('accept', 'json')
+      .then((result): GausDeviceConfiguration => result.body);
   }
 
-  checkForUpdates(deviceAuthParameters: GausDeviceAuthParameters): Promise<GausUpdate[]> {
+  checkForUpdates(
+    deviceAuthParameters: GausDeviceAuthParameters,
+    updateParameters: GausUpdateParameter[]
+  ): Promise<GausUpdate[]> {
+    if (!updateParameters) {
```
Comment (toshi38) [inline src/gaus-client.ts]: Hmm... is this really required? I mean an empty array is valid (ie no parameters to report) (I think) ... so do we need this or can we just default it?

File: package.json (hunk 1 above)
Comment (AndreeFrank) [reply package.json]: Done

File: package.json (hunk 2 above)
Comment (AndreeFrank) [reply package.json]: Done

File: src/gaus-client.ts (hunk 3 above)
Comment (AndreeFrank) [reply src/gaus-client.ts]: Makes sense to default it to empty array

**Pull request merged**